### ISSUE TITLE
Incorrect multiplication tables

### ISSUE THREAD

**Issue by flsmith** [gap-packages/smallsemi#2]
When trying to get the multiplication table of a small semigroup, I can trigger this:

```
gap> S := AsSemigroup(IsTransformationSemigroup, SmallSemigroup(4, 18));
<transformation semigroup of size 4, degree 5 with 3 generators>
gap> sortedlist := AsSSortedList(S);;
gap> M := MultiplicationTable(S);;
gap> for i in [1 .. 4] do
> for j in [1 .. 4] do
> if not sortedlist[i] * sortedlist[j] = sortedlist[M[i][j]] then
> Error(i, j, M[i][j]);
> fi;
> od;
> od;
Error, 141 at *stdin*:40 called from
<function "unknown">( <arguments> )
 called from read-eval loop at *stdin*:43
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
```
but the following is fine:

```
gap> S := Semigroup(GeneratorsOfSemigroup(S));
<transformation semigroup of degree 5 with 3 generators>
gap> sortedlist := AsSSortedList(S);;
gap> M := MultiplicationTable(S);;
gap> for i in [1 .. 4] do
> for j in [1 .. 4] do
> if not sortedlist[i] * sortedlist[j] = sortedlist[M[i][j]] then
> Error(i, j, M[i][j]);
> fi;
> od;
> od;
gap> 
```

**Comment by fingolfin** [comment]
So, is there a chance somebody will submit a fix to smallsemi, and a new release with this fixed will be released?

**Comment by james-d-mitchell** [comment]
Yes I intend to do this soon.

**Comment by olexandr-konovalov** [comment]
Fixed in #7 by @james-d-mitchell
